src/pages/cli/project/troubleshooting.mdx

```diff
@@ -0,0 +1,263 @@
+In the AWS CloudFormation console, search for your application's stack name. For instance, if your application was named "lil" and deployed in your "dev" environment, 
+Amplify would have generated a CloudFormation stack named `amplify-lil-dev-{random chars}`.
+In the Cloudformation console, click on the application's stack name and then click on the "Resources" tab. 
+The "Resources" tab displays the deployment status of all the resources in the stack. It will also show the deployment error reason. 
+Refer the [CloudFormation troubleshooting document](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/troubleshooting.html#basic-ts-guide) to gain insight into the error messages.
+
+**Note:**
+To debug any issues related to secrets not being correctly copied across environments, refer to [Multi-environment flows](/cli/function/secrets/#multi-environment-flows) section in the [Access secret values](/cli/function/secrets) documentation. 
+
+![CloudFormation status logs for a deployment failure of a resource](/images/ts-guide-cfn-error.png)
+
+### Push failures from resources exceeding AWS service quotas
+<Callout information>
+
+**Errors:** 
+
+"Limit on the number of resources in a single stack operation exceeded”  
+"Cannot exceed quota for PoliciesPerRole"
+</Callout>
+
+AWS maintains [service quotas](/general/latest/gr/aws-service-information.html) for various AWS services. If your application contains many resources, such as large GraphQL schemas with many resolvers or many REST APIs, you may hit a service limit.
+To identify if a push failure is because of exceeding [AWS Service level quotas](/general/latest/gr/aws-service-information.html) check the AWS console for the CloudFormation deployment/failure logs.
+
+**Suggested Resolution:** 
+1. If your application's resource requirements are impaired by AWS service-quotas, you can use the [AWS Service Quotas console](https://console.aws.amazon.com/servicequotas) to view and request increases for most AWS quotas.
+1. If you are using the Amplify DataStore, refer to [Amplify DataStore Best Practices](/whitepapers/latest/amplify-datastore-implementation/amplify-datastore-best-practices.html) for insights into making the correct design decisions and optimize resource allocation in your application. 
+1. If the service quotas cannot be increased, then this may require you to redesign some aspects of your application for sustainable scaling in a serverless environment. Referring to [AWS Knowledge Center](https://aws.amazon.com/premiumsupport/knowledge-center/) may help understand the different prescribed design changes. For example, a complex GraphQL schema could be simplified to reduce the number of GraphQL resources in use e.g collapsing models, composite keys, Lambda resolvers. Using escape hatches like overrides and export may help to scale the application beyond the CLI generated architecture. 
```

Comment:
"e.g collapsing models, composite keys, Lambda resolvers." -> "For example, ..." I don't fully get what "collapsing composite keys or collapsing Lambda resolvers" means.

Reply:
Link for overrides and export would be great.